CMakeLists.txt

```diff
@@ -416,11 +424,11 @@ ExternalProject_Add(${PNG_TARGET}
        BINARY_DIR ${BUILD_DIR}/png_build
        INSTALL_DIR ${CMAKE_INSTALL_PREFIX}
        # CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR>
-       CONFIGURE_COMMAND ${CMAKE_COMMAND} ${CMAKE_CORE_BUILD_FLAGS} ${ZLIB_CMAKE_FLAGS} -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR> <SOURCE_DIR>
+       CONFIGURE_COMMAND ${CMAKE_COMMAND} -DPNG_ARM_NEON=off ${CMAKE_CORE_BUILD_FLAGS} ${ZLIB_CMAKE_FLAGS} -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR> <SOURCE_DIR>
```

Comment:
And setting it to `on` when building on UNIX does not have problems?

Reply:
You're right, I missed that this is common code with x86.

Reply:
so maybe have a variable set like
```cmake
if(IF(${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm") )
    set(AV_PNG_ARM_NEON on)
else()
    set(AV_PNG_ARM_NEON off)
endif()
```
so we can put `-DPNG_ARM_NEON=${AV_PNG_ARM_NEON}`
What do you think?

Reply:
That would make sense. I didn't suggest it because there're issues with CMAKE_SYSTEM_PROCESSOR - https://gitlab.kitware.com/cmake/cmake/-/issues/22881. Thinking about this more I think we should be safe with your approach because if the detection fails then we'll just use `set(AV_PNG_ARM_NEON off)`.